modules/apps/translation/translation-web/src/main/java/com/liferay/translation/web/internal/portlet/action/ImportTranslationMVCActionCommand.java

```diff
@@ -152,6 +153,22 @@ protected void doProcessAction(
 				MultiSessionMessages.add(
 					actionRequest, portletResource + "requestProcessed");
 			}
+
+			actionRequest.setAttribute(
+				WebKeys.REDIRECT,
+				_getRedirect(
+					classNameId, classPK, groupId, actionResponse,
+					ParamUtil.getString(actionRequest, "redirect")));
+
+			MultiSessionMessages.add(
```

Comment:
This is wrong. Use the request.getSession() session; the class you're using is for success/error mesages, not for general data (although it may work).